base/set.jl

```diff
@@ -141,6 +141,27 @@ function unique(f::Callable, C)
     out
 end
 
+"""
+    alldistinct(itr)
+
+Test if all values from `itr` are distinct, when compared with `isequal`.
+"""
+function alldistinct(C)
+    seen = Set{eltype(C)}()
+    for x in C
+        if in(x, seen)
+            return false
+        else
+            push!(seen, x)
+        end
+    end
+    true
+end
+
+alldistinct(::Set) = true
+
+alldistinct(r::Range) = (length(r) == 1) || (step(r) > 0)
```

Comment:
This should probably be:

    (step(r) != 0) || (length(r) <= 1)

or potentially:

    alldistinct{T}(r::Range{T}) = (step(r) !== zero(T)) || (length(r) <= one(T))

and add tests for negative step and zero length ranges.